### PR TITLE
cleanup: remove unused overwrite variable

### PR DESCRIPTION
**Pull request by ptab**

### CODE REVIEW

File: tests/test_resources/test_data.py [hunk 1]
```diff
@@ -152,14 +152,6 @@ def get_project_with_stages(
...
-def assert_roundtrip(file_path: Path, actual_contents: str, overwrite: bool = False):
-    if overwrite:
-        if not file_path.exists():
-            os.makedirs(file_path.parent, exist_ok=True)
-
-        with open(file_path, "w+", encoding="utf-8") as file:
-            file.write(actual_contents)
-            assert not overwrite, "Should not commit with overwrite"
-
+def assert_roundtrip(file_path: Path, actual_contents: str):
```
[the remark this answers is not in the document]
Comment (ptab) [reply tests/test_resources/test_data.py:155]: I don't get it. If we're using the same code to generate a file to compare against, then what's the point of having a test?

[the remark this answers is not in the document]
Comment (Jorg88) [reply tests/test_resources/test_data.py:155]: Well several reasons

- it's only done when you explicitly tell your test to do so
- it's much faster than manual editing
- sometimes it's basically impossible to manually create the same result/string/markdown/file/pretty-print/whatever as the code generates
- the test still validates bugs when you don't expect a change in your result

I'm pretty sure we apply this same principle in our scala code, don't we?

[the remark this answers is not in the document]
Comment (ptab) [reply tests/test_resources/test_data.py:155]: I really hope we don't 😅 This sounds to me like snapshot testing, where pretty much everyone I've ever seen use it in practice follows this workflow:
1. run tests
2. test fails with a change from the previous snapshot
3. update snapshot with the new value
4. test is green, yay! 🤷 
5. commit and proceed

[the remark this answers is not in the document]
Comment (ptab) [reply tests/test_resources/test_data.py:155]: To be honest, I'm not invested enough in this change to contest and I'm happy to close the PR). I do think that having the same code under test automatically generate the files used to validate it significantly reduces the value of the test.

**Comment by ptab** [comment]
Turns out this flag is still used in some use cases, so for the time being I'll close this PR.

**Pull request closed without merging**